int parameter in `{.global.}` assignment generates invalid code

### Description

Procedure that accepts int/bool and then assigns that to a global variable generates invalid code, strings work fine.

Example:
```Nim
proc example(a = 1) =
  let b {.global.} = a

example()
```

### Nim Version

bdccc9fef93d6598ec453a19152c4e98539f783f
Also occurs on 1.2, 1.4, 1.6; arc, refc, markandsweep, none

### Current Output

```text
......................................................................
/home/user/test.nim(2, 7) Hint: 'b' is declared but not used [XDeclaredButNotUsed]
CC: .nim/devel/lib/std/private/digitsutils.nim
CC: .nim/devel/lib/system/dollars.nim
CC: .nim/devel/lib/system.nim
CC: test.nim
/home/user/.cache/nim/test_d/@mtest.nim.c: In function ‘NimMainModule’:
/home/user/.cache/nim/test_d/@mtest.nim.c:122:22: error: ‘a_p0’ undeclared (first use in this function)
  122 |         b__test_u3 = a_p0;
      |                      ^~~~
/home/user/.cache/nim/test_d/@mtest.nim.c:122:22: note: each undeclared identifier is reported only once for each function it appears in
Error: execution of an external compiler program 'gcc -c  -w -fmax-errors=3 -pthread -g3 -Og   -I/home/user/.nim/devel/lib -I/home/user -o /home/user/.cache/nim/test_d/@mtest.nim.c.o /home/user/.cache/nim/test_d/@mtest.nim.c' failed with exit code: 1
```


### Expected Output

_No response_

### Possible Solution

_No response_

### Additional Information

_No response_

--- a/compiler/semstmts.nim
+++ b/compiler/semstmts.nim
@@ -563,7 +563,7 @@ proc semVarMacroPragma(c: PContext, a: PNode, n: PNode): PNode =
         return result
 
 template isLocalSym(sym: PSym): bool =
-  sym.kind in {skVar, skLet} and not
+  sym.kind in {skVar, skLet, skParam} and not
     ({sfGlobal, sfPure} * sym.flags != {} or
       sfCompileTime in sym.flags) or
       sym.kind in {skProc, skFunc, skIterator} and

--- a/tests/global/t21896.nim
+++ b/tests/global/t21896.nim
@@ -1,0 +1,9 @@
+discard """
+  errormsg: "cannot assign local to global variable"
+  line: 7
+"""
+
+proc example(a:int) =
+  let b {.global.} = a
+
+example(1)

